### ISSUE TITLE
[MODULES]: Load modules asynchronously

### ISSUE THREAD

**Issue by MarcusTXK** [source-academy/modules#79]
# Feature

Is your feature request related to a problem? Please describe.

We need to load the module asynchronously and somehow indicate to the user that module loading is in progress. This should be analogous to Sling sending a program to the robot.  

Describe the solution you'd like

When there is a module load, we should do the following upon Run:
(1) indicate that module loading is in progress
(2) load the module(s) asynchronously
(3) when (2) is successful, indicate that module loading is done
(3) run the program

While (2) is under way, the user should be able to interact with the system.

## Checklist

- [x] I checked the requested feature does not already exist
- [x] I checked to make sure that this issue has not already been filed

**Comment by martin-henz** [comment]
Yes, thanks for writing this up. I agree that we should have this feature. Any plans for implementing it?

**Comment by martin-henz** [comment]
What is the status of this feature? I think I just encountered an issue related to this, when playing with the SA@NUS deployed to staging. I  uploaded the mission M3: Curve Introduction. After entering the solution of the first question, I pressed "Run" and didn't see any output. There was no error in the console.

The second time I tried, it worked. This may cause hickups in CS1101S.

**Comment by martin-henz** [comment]
Here is a more detailed proposal:
"Run" with import statements:

Case 1: all imported modules are available locally (cached from earlier runs)
normal behavior. no extra messages

Case 2: at least one imported module is not available locally
Consequences:
* The program is not run: instead the system becomes responsive immediately after pressing "Run"
* The REPL shows: "Loading modules x, y, z from server..."

Case 2a: all non-locally-available modules can be loaded within a timeout of say 10 seconds
Consequences:
* The REPL displays: "Modules x, y, z loaded from server. Press Run again."

Case 2b: not all non-locally-available modules can be loaded within a timeout of say 10 seconds
Consequences:
* The REPL displays: "Modules x, y, z could not be loaded from server. Check your network connection and try again."
